### ISSUE TITLE
Add Model Details for Pipeline API

### ISSUE THREAD

**Issue by dhruvjimulia-sys** [huggingface/transformers#12845]
# 🚀 Feature request

I think it will be great if we could add further details about what particular model the Pipeline API is using. In particular, we can add the following details relating to the models in a readable markdown file:
* Dataset used for training
* Type of transformer used
* Hyperparameters and neural net architecture
* Evaluation metrics: accuracy + precision + recall on test portion of dataset
## Motivation
I was working on different sentiment analysis datasets with different transformers when I stumbled across the HuggingFace pipeline API. I was impressed by the performance of the models, but I wasn't quantitively sure about the accuracy of the model, neither was I sure about how it was trained.

I believe that many other users of the Pipeline API will also benefit from such a change.

P.S. If there is some source of such information and I've missed it, please let me know

**Comment by NielsRogge** [comment]
This can be interesting. Are you interested in opening a PR? You could add a section with model details for every pipeline by editing the documentation pages (i.e. the .rst files).

**Comment by dhruvjimulia-sys** [comment]
Yes I would like to try opening a PR for this. However, it will take me some time to understand the code behind how the pipeline works.

**Comment by NielsRogge** [comment]
cc @Narsil, I think it would be very beneficial to document which model is used by default for each pipeline.

**Comment by Narsil** [comment]
Should we do that at runtime with a log that we chose a default model, or you had something more static (definitely doable, but probably tedious to maintain.)

**Comment by dhruvjimulia-sys** [comment]
Yes I've been busy lately - although I thought I could examine the code, it has not been possible given my schedule. Given that this is an important problem others are facing too, feel free to open a PR.

**Comment by dhruvjimulia-sys** [comment]
@Narsil I personally feel that having a runtime log will be more than enough given the ease at which one can get started with using the pipeline API - one can simply implement it to find out what model is working behind the scenes
